### DISCUSSION TITLE
Health checks for individual cameras

### DISCUSSION THREAD

**Discussion by brujoand** [blakeblackshear/frigate#8174] [General]
Once a month or so I'll have one camera stream where "something went wrong" and the stream is broken. Not quite sure what the cause is yet, but a restart of Frigate solves it. But this got me thinking that I should have health check for each individual camera. Would I need to do a ffprobe on each stream to verify this or is there some built in support in frigate? Maybe the go2rtc api could tell me that the stream is down?

**Comment by NickM-27** [comment]
Frigate ready has a watchdog and restarts ffmpeg if the stream fails. 

When this happens frigate sets the camera_fps value in stats to 0 for that camera. 

With the new home assistant integration version 5.0 the camera will also be marked as unavailable
